### PR TITLE
Add huge checkbox

### LINKED ISSUE #3146: [Build] Custom Huge Checkbox Input - Vanilla
**User Story:**
As a Vanilla Spark user I want to see a custom Checkbox Input type added to Spark.

**Notes:**
- This is for Vanilla
- Get Link From Jane, Sketch link wouldn't load during planning.

**AC:**
Huge Checkmark Input will have
- A Normal state
- A Checked/Selected state
- A Hover state
- A Disabled state
- Max characters per input? (Max Size, text can not wrap)
- Input will have a Min Width of (JANE)
- Desktop: will see 5 stack pattern combinations 
- Stack on Mobile will be Vertical
- will see Stack Pattern Combination visual in SB

image

image

### PR DESCRIPTION
**Pull request by RVMendoza**
## What does this PR do?
Add a huge version of checkbox
image

### Associated Issue
Fixes #3146   

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [x] Update Spark Docs

### Code
- [x] Build Component in HTML

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Design Review
 - [x] Design reviewed and approved

### CODE REVIEW

File: html/settings/_settings.scss [hunk 1]
```diff
@@ -1059,6 +1059,71 @@ $sprk-checkbox-custom-input-cursor-disabled: default !default;
...
+/// The max-width of the huge checkbox input.
+$sprk-checkbox-input-huge-max-width: $sprk-input-container-huge-max-width !default;
+/// The transition of the huge checkbox input.
+$sprk-checkbox-huge-container-transition: $sprk-checkbox-transition !default;
+/// The border settings of the huge checkbox input.
+$sprk-checkbox-huge-container-border: 2px $sprk-black-tint-50 solid !default;
+/// The border-radius of the huge checkbox input.
+$sprk-checkbox-huge-container-border-radius: 4px !default;
+/// The padding of the huge checkbox input.
+$sprk-checkbox-huge-container-padding: 20px 16px 20px 55px !default;
```
Comment (afebbraro) [inline html/settings/_settings.scss:1074]: Lets use our spacing vars here where we can, can use space-m in place of that 16px

File: html/settings/_settings.scss [hunk 2]
```diff
@@ -1059,6 +1059,71 @@ $sprk-checkbox-custom-input-cursor-disabled: default !default;
...
+/// The border color of the huge checkbox input when focused and hovered.
+$sprk-checkbox-huge-container-focus-hover-border-color: $sprk-black !default;
+/// The border color of the huge checkbox input when checked.
+$sprk-checkbox-huge-container-checked-border-color: $sprk-green !default;
+/// The border color of the huge checkbox input when checked and hovered.
+$sprk-checkbox-huge-container-checked-hover-border-color: $sprk-black !default;
+/// The border color of the huge checkbox input when checked and focused.
+$sprk-checkbox-huge-container-focus-checked-border-color: $sprk-black !default;
+/// The width of the custom checkbox element of the huge checkbox input.
+$sprk-checkbox-huge-custom-input-height: 1.5rem !default;
```
Comment (afebbraro) [inline html/settings/_settings.scss:1102]: the comment here says the width but the var name says height. Which one is it?

File: html/settings/_settings.scss [hunk 3]
```diff
@@ -1059,6 +1059,71 @@ $sprk-checkbox-custom-input-cursor-disabled: default !default;
...
+/// The height of the custom checkbox element of the huge checkbox input.
+$sprk-checkbox-huge-custom-input-weight: 1.5rem !default;
```
Comment (afebbraro) [inline html/settings/_settings.scss:1104]: the comment says height but the var name says weight

File: src/data/sass-vars.json [hunk 4]
```diff
@@ -0,0 +1,19903 @@
+[
+  {
```
Comment (afebbraro) [inline src/data/sass-vars.json:2]: Thought we added these files to the gitignore?

File: src/pages/using-spark/components/input.mdx [hunk 5]
```diff
@@ -92,6 +93,35 @@ than one option from a list.
...
+- Recommend content to not exceed 40 characters.
+- The inputs become stacked in a grid. Stacking will
+take place automatically based on options presented.
+- Up to 6 options can be presented, an alternative
+to use is the Select Box for greater than 6 options.
+
+### Anatomy
+- Must have an associated question.
+- Must contain concise body text for each option.
+- Width of each Huge Radio should be the same size.
```
Comment (afebbraro) [inline src/pages/using-spark/components/input.mdx]: ``
```suggestion
- Width of each Huge Checkbox should be the same size.
```

File: src/pages/using-spark/components/input.mdx [hunk 6]
```diff
@@ -92,6 +93,35 @@ than one option from a list.
   hasHTML
 />
 
+### Huge Checkbox
+[Back to Variants](#variants)
+
+The Huge Checkbox component is intended for flows
+that present single questions in a step by step process
```
Comment (afebbraro) [inline src/pages/using-spark/components/input.mdx]:
```suggestion
that present single questions in a step by step process.
```

File: src/pages/using-spark/components/input.mdx [hunk 7]
```diff
@@ -92,6 +93,35 @@ than one option from a list.
...
+For this reason, Huge Checkbox should only be
+used for stand-alone question flows.
+
+#### Guidelines
+- Ensure list is in logical order for inputs (groups will stack).
+- Options are short and concise.
+- Recommend content to not exceed 40 characters.
```
Comment (afebbraro) [inline src/pages/using-spark/components/input.mdx]: ``
```suggestion
- Content should not exceed 40 characters.
```

File: src/pages/using-spark/components/input.mdx [hunk 8]
```diff
@@ -92,6 +93,35 @@ than one option from a list.
...
+- The inputs become stacked in a grid. Stacking will
+take place automatically based on options presented.
```
Comment (afebbraro) [inline src/pages/using-spark/components/input.mdx]:
```suggestion
take place automatically if you use the layout code based on options presented.
```

We need to make sure they know the automatically part only happens if they use the stack code.

File: src/pages/using-spark/components/input.mdx [hunk 9]
```diff
@@ -92,6 +93,35 @@ than one option from a list.
...
+to use is the Select Box for greater than 6 options.
+
+### Anatomy
+- Must have an associated question.
+- Must contain concise body text for each option.
+- Width of each Huge Radio should be the same size.
+
+<ComponentPreview
+  componentName="input-huge-checkbox--default-story"
+  hasReact
```
Comment (afebbraro) [inline src/pages/using-spark/components/input.mdx]: We dont have the react or angular ones yet.

File: src/data/sass-vars.json (hunk 4 above)
Comment (RVMendoza) [reply src/data/sass-vars.json:2]: I thought so too, I believe it came through when I pulled in staging

File: src/pages/using-spark/components/input.mdx (hunk 5 above)
Comment (RVMendoza) [reply src/pages/using-spark/components/input.mdx]: Fixed!

File: html/base/inputs/_checkbox.scss [hunk 10]
```diff
@@ -82,3 +80,101 @@
...
+          border-color: $sprk-checkbox-huge-custom-input-checkbox-element-checked-border-color;
+        }
+      }
+
+      // Styles for container when checkbox is focused on while being checked
+      &:focus + .sprk-b-Checkbox__label {
+        border-color: $sprk-checkbox-huge-container-focus-checked-border-color;
+      }
+    }
+
```
Comment (afebbraro) [inline html/base/inputs/_checkbox.scss:178]: extra line

File: html/settings/_settings.scss [hunk 11]
```diff
@@ -1059,6 +1059,71 @@ $sprk-checkbox-custom-input-cursor-disabled: default !default;
...
+$sprk-checkbox-huge-checkmark-icon-height: 0.5rem !default;
+/// The width of the checkmark inside of the huge checkbox input.
+$sprk-checkbox-huge-checkmark-icon-width: .75rem !default;
+/// The left value of the checkmark inside of the huge checkbox input.
+$sprk-checkbox-huge-checkmark-icon-left: 22px !default;
+/// The top value of the checkmark inside of the huge checkbox input.
+$sprk-checkbox-huge-checkmark-icon-top: 22px !default;
+
+
+
```
Comment (afebbraro) [inline html/settings/_settings.scss:1126]: only need one extra line here

File: html/settings/_settings.scss [hunk 12]
```diff
@@ -1059,6 +1059,67 @@ $sprk-checkbox-custom-input-cursor-disabled: default !default;
...
+/// The width of the custom checkbox element of the huge checkbox input.
+$sprk-checkbox-huge-custom-input-width: 1.5rem !default;
+/// The top value of the custom checkbox element of the huge checkbox input.
+$sprk-checkbox-huge-custom-input-top: $sprk-space-m !default;
+/// The left value of the custom checkbox element of the huge checkbox input.
+$sprk-checkbox-huge-custom-input-left: $sprk-space-m !default;
+/// The border color of the custom checkbox element of the huge checkbox input when hovered.
+$sprk-checkbox-huge-custom-input-hover-border-color: $sprk-black !default;
+/// The border color of the custom checkbox element of the huge checkbox input when disabled.
+$sprk-checkbox-huge-custom-input-disabled-border-color: $sprk-black-tint-25 !default;
```
Comment (afebbraro) [inline html/settings/_settings.scss:1110]:
```suggestion
$sprk-checkbox-huge-custom-input-disabled-border-color: $sprk-black-tint-50 !default;
```

**Pull request merged**